### PR TITLE
impl `getData` trait for boolean/string column & improve column

### PR DESCRIPTION
**Pull request by Veeupup**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- [x] impl `getData` for BooleanColumn
- [ ] impl `getData` for StringColumn
- [ ] impl `Debug` for Column

## Changelog

- Improvement
- 
## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/datavalues2/src/columns/boolean/iterator.rs [hunk 1]
```diff
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::sync::Arc;
+use std::sync::RwLock;
```
Comment (PsiACE) [inline common/datavalues2/src/columns/boolean/iterator.rs:16]: `common_infallible::RwLock` or `parking_lot::RwLock`

File: common/datavalues2/src/columns/boolean/mod.rs [hunk 2]
```diff
@@ -99,6 +124,8 @@ impl Column for BooleanColumn {
         unsafe {
             Arc::new(Self {
                 values: self.values.clone().slice_unchecked(offset, length),
```
Comment (sundy-li) [inline common/datavalues2/src/columns/boolean/mod.rs:126]: It's false cached, so we do not need to clone the original slices?

File: common/datavalues2/src/columns/boolean/iterator.rs (hunk 1 above)
Comment (Veeupup) [reply common/datavalues2/src/columns/boolean/iterator.rs:16]: got it.

File: common/datavalues2/src/columns/boolean/mod.rs (hunk 2 above)
Comment (Veeupup) [reply common/datavalues2/src/columns/boolean/mod.rs:126]: if not cached data, we do not need to copy the data, but if data cached, it seems that we need to copy the underlying data and make `data_cached` true.

**Comment by Veeupup** [comment]
scalar type need to refactor.

**Pull request closed without merging**